nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/tools/PeakSlider.java

```diff
@@ -173,26 +175,61 @@ public void initSlider(VBox vBox) {
         Pane filler5 = new Pane();
         HBox.setHgrow(filler5, Priority.ALWAYS);
 
-        if (sliderToolBar != null) {
-            sliderToolBar.getItems().add(closeButton);
-            sliderToolBar.getItems().add(filler1);
-            sliderToolBar.getItems().add(actionMenu);
-            sliderToolBar.getItems().addAll(buttons);
-            sliderToolBar.getItems().add(filler2);
-            sliderToolBar.getItems().addAll(atomXFieldLabel, atomXLabel, filler3, atomYFieldLabel, atomYLabel, filler4, intensityFieldLabel, intensityLabel);
-            sliderToolBar.getItems().add(filler5);
-
-            // The different control items end up with different heights based on font and icon size,
-            // set all the items to use the same height
-            sliderToolBar.heightProperty().addListener((observable, oldValue, newValue) -> GUIUtils.toolbarAdjustHeights(List.of(sliderToolBar)));
+        sliderToolBar.getItems().add(closeButton);
+        sliderToolBar.getItems().add(filler1);
+        sliderToolBar.getItems().add(actionMenu);
+        sliderToolBar.getItems().addAll(buttons);
+        sliderToolBar.getItems().add(filler2);
+        sliderToolBar.getItems().addAll(atomXFieldLabel, atomXLabel, filler3, atomYFieldLabel, atomYLabel, filler4, intensityFieldLabel, intensityLabel);
+        sliderToolBar.getItems().add(filler5);
+
+        // The different control items end up with different heights based on font and icon size,
+        // set all the items to use the same height
+        sliderToolBar.heightProperty().addListener((observable, oldValue, newValue) -> GUIUtils.toolbarAdjustHeights(List.of(sliderToolBar)));
+
+        // Setup listeners
+        selectedPeaksListener = event -> setActivePeaks(controller.selPeaks.get());
+        controller.selPeaks.addListener(selectedPeaksListener);
+        for (PolyChart chart : controller.getCharts()) {
+            addKeyBindingsToChart(chart);
         }
-        controller.selPeaks.addListener(e -> setActivePeaks(controller.selPeaks.get()));
+        ((ObservableList<PolyChart>) controller.getCharts()).addListener(chartsListener);
+   }
+
+    /**
+     * Add key bindings to newly added charts.
+     * @param change The change to the FXMLController charts list.
+     */
+    private void updateKeyBindings(ListChangeListener.Change<? extends PolyChart> change) {
+        if (change.next()) {
+            for (PolyChart chart: change.getAddedSubList()) {
+                addKeyBindingsToChart(chart);
+            }
+        }
+    }
```

Comment:
added this listener so if a user added another chart while the peak slider was opened, the new chart would also have the peak slider key bindings set.